hedera-node/src/main/java/com/hedera/services/state/virtual/schedule/ScheduleVirtualValue.java

```diff
@@ -0,0 +1,635 @@
+		return equalityHash().asLong();
+	}
+
+	public String equalityCheckValue() {
+		return equalityHash().toString();
+	}
+
+	private HashCode equalityHash() {
+		// if this changes at all, the equality database for scheduled transactions will need to be re-built
+		return buildEqualityHash(
+				memo != null ? memo.getBytes(StandardCharsets.UTF_8): new byte[] {},
+				grpcAdminKey != null ? grpcAdminKey.toByteArray() : new byte[] {},
+				scheduledTxn.toByteArray(),
+				expirationTimeProvided != null ? expirationTimeProvided.toGrpc().toByteArray() : new byte[] {},
+				waitForExpiryProvided ? new byte[] {1} : new byte[] {0});
+	}
+
+	@Override
+	public String toString() {
+		var helper = MoreObjects.toStringHelper(ScheduleVirtualValue.class)
+				.add("scheduledTxn", scheduledTxn)
+				.add("expirationTimeProvided", expirationTimeProvided)
+				.add("calculatedExpirationTime", calculatedExpirationTime)
+				.add("executed", executed)
+				.add("waitForExpiryProvided", waitForExpiryProvided)
+				.add("calculatedWaitForExpiry", calculatedWaitForExpiry)
+				.add("deleted", deleted)
+				.add("memo", memo)
+				.add("payer", readablePayer())
+				.add("schedulingAccount", schedulingAccount)
+				.add("schedulingTXValidStart", schedulingTXValidStart)
+				.add("signatories", signatories.stream().map(CommonUtils::hex).toList())
+				.add("adminKey", describe(adminKey))
+				.add("resolutionTime", resolutionTime);
+		return helper.toString();
+	}
+
+	private String readablePayer() {
+		return Optional.ofNullable(effectivePayer()).map(EntityId::toAbbrevString).orElse("<N/A>");
+	}
+
+	@Override
+	public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+		int n = in.readInt();
+		bodyBytes = new byte[n];
+		in.readFully(bodyBytes);
+		if (in.readByte() == 1) {
+			calculatedExpirationTime = new RichInstant(in.readLong(), in.readInt());
+		} else {
+			calculatedExpirationTime = null;
+		}
+		calculatedWaitForExpiry = in.readByte() == 1;
+		executed = in.readByte() == 1;
+		deleted = in.readByte() == 1;
+		if (in.readByte() == 1) {
+			resolutionTime = new RichInstant(in.readLong(), in.readInt());
+		} else {
+			resolutionTime = null;
+		}
+		int k = in.readInt();
+		for (int x = 0; x < k; ++x) {
+			n = in.readInt();
+			byte[] bytes = new byte[n];
+			in.readFully(bytes);
+			witnessValidSignature(bytes);
+		}
+
+		initFromBodyBytes();
+	}
+
+	@Override
+	public void serialize(SerializableDataOutputStream out) throws IOException {
+		out.writeInt(bodyBytes.length);
+		out.write(bodyBytes);
+		if (calculatedExpirationTime == null) {
+			out.writeByte((byte) 0);
+		} else {
+			out.writeByte((byte) 1);
+			out.writeLong(calculatedExpirationTime.getSeconds());
+			out.writeInt(calculatedExpirationTime.getNanos());
+		}
+		out.writeByte((byte) (calculatedWaitForExpiry ? 1 : 0));
+		out.writeByte((byte) (executed ? 1 : 0));
+		out.writeByte((byte) (deleted ? 1 : 0));
+		if (resolutionTime == null) {
```

Comment:
`resolutionTime == null` should be true iff `!(executed || deleted)` so this byte seems redundant, but hardly important ofc.